lib/compile/formats.js

```diff
@@ -90,6 +90,8 @@ function date(str) {
 
 
 function time(str, full) {
+  if (str === undefined) return false;
```

Comment:
The guards should be on the calling side, in date_time function. This function is also used for "time" format and it will never be called for undefined, so this condition is only necessary if called from date_time